### PR TITLE
Reuse object_shader for "wielditem" and "item" entity drawtypes

### LINKED ISSUE #9525: Falling nodes glow in the dark
##### Minetest version
```
5c1b72544a6158346b7d8608c895678fe7d5a19b
```

##### Summary
Falling nodes glow in the dark.

##### Steps to reproduce
1. Fly into a cave that is pitch black
2. Place any falling node like sand

### PR DESCRIPTION
**Pull request by dcbrwn**
Fixes #9525. But the fix is controversial: I removed wielditem shader and reused object shader instead. This change should be tested more thoroughly contrary to one suggested by @sfan5. If this will block PR for more than week or two, I'll fallback to the suggested quick fix.

Ready for Review.

## How to test

Original bug can be tested with minetest game: go into a dark cave and place sand block. Before this fix, sand block would glow. Now it shouldn't.

### CODE REVIEW

**Comment by dcbrwn** [comment]
Oops. Need to rebase to minetest's master.

**Comment by dcbrwn** [comment]
Extruded and cube items work fine. When shaders are disabled, meshes are still post-processed to emulate lighting. This differs a bit from what object_shader does but not too much.

**Comment by paramat** [comment]
As 5.3.0 is planned to follow shortly after 5.2.0 i think i prefer we just use the simple workaround for release, to reduce risk and reduce release workload. Seems to be no loss as shaders are not yet applied to wieldmesh anyway.
But, see what other core devs think about this first.

**Comment by sfan5** [comment]
I agree, the simple workaround should go into 5.2.0, this PR into 5.3.0.

**Comment by dcbrwn** [comment]
Fair enough. I'll open PR with alternative fix.

**Comment by SmallJoker** [comment]
@dcbrwn 5.2.0 is out now. Would you please be so nice and rebase this PR one more time?

**Comment by dcbrwn** [comment]
Seems to be working properly after rebase

**Comment by SmallJoker** [comment]
Switching time to "night":

```C++
Thread 1 "minetest" received signal SIGSEGV, Segmentation fault.
0x00005555557b3b90 in WieldMeshSceneNode::setNodeLightColor(irr::video::SColor) ()
(gdb) bt
#0  0x00005555557b3b90 in WieldMeshSceneNode::setNodeLightColor(irr::video::SColor) ()
#1  0x00005555556dbba4 in GenericCAO::setNodeLight(unsigned char) ()
#2  0x00005555556dbe86 in GenericCAO::updateLight(unsigned char) ()
#3  0x00005555556ac4fc in ClientEnvironment::addActiveObject(ClientActiveObject*) ()
#4  0x00005555556ac58a in ClientEnvironment::addActiveObject(unsigned short, unsigned char, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) ()
#5  0x0000555555941807 in Client::handleCommand_ActiveObjectRemoveAdd(NetworkPacket*) ()
#6  0x000055555569e92d in Client::ProcessData(NetworkPacket*) ()
#7  0x000055555569eb8e in Client::ReceiveAll() ()
#8  0x00005555556a4884 in Client::step(float) ()
#9  0x00005555557147cb in Game::run() ()
#10 0x0000555555715887 in the_game(bool*, bool, InputHandler*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned short, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, ChatBackend&, bool*, SubgameSpec const&, bool) ()
#11 0x00005555556b58c2 in ClientLauncher::run(GameParams&, Settings const&) ()
#12 0x000055555568a5ff in main ()
```

File: src/client/content_cao.cpp [hunk 1]
```diff
@@ -826,11 +826,11 @@ void GenericCAO::setNodeLight(u8 light)
 	video::SColor color(255, light, light, light);
 
 	if (m_prop.visual == "wielditem" || m_prop.visual == "item") {
-		// Since these types of visuals are using their own shader
-		// they should be handled separately
-		if (m_wield_meshnode)
-			m_wield_meshnode->setColor(color);
-	} else if (m_enable_shaders) {
+		m_wield_meshnode->setNodeLightColor(color);
```
Comment (SmallJoker) [inline src/client/content_cao.cpp:829]: `if (m_wield_meshnode)` missing.

Comment (sfan5) [reply src/client/content_cao.cpp:829]: This code should also be moved into the `if (m_enable_shaders)` / `else` below.

Comment (dcbrwn) [reply src/client/content_cao.cpp:829]: Oh, I wasn't expecting that.

@sfan5 no, setNodeLightColor handles m_enable_shaders by itself. Given that wield items setup materials by themselves I decided play along and do it this way.

**Pull request merged**